richard-kendra/server.js

```diff
@@ -0,0 +1,195 @@
+  // The following line of code matches #5 in the full-stack-diagram.png. It appears that no methods in article.js interact with this piece of server.js. This piece also is not a part of CRUD, because it's just serving the new.html page and does not do anything with the app's data in the database.
+  response.sendFile('new.html', {root: './public'});
+});
+
+
+// REVIEW: Routes for making API calls to use CRUD Operations on our database
+app.get('/articles', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This line of code matches #3 & #5 in the diagram, which is making a query to the database and sending a response. This interacts with the method fetchAll() in article.js, which asks for all the article data. This corresponds to the "R" in CRUD, because it's retrieving data from the database.
+  client.query('SELECT * FROM articles;')
+    .then(function(result) {
+      response.send(result.rows);
+    })
+    .catch(function(err) {
+      console.error(err)
+    })
+});
+
+app.post('/articles', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This matches #3 & #5 in the diagram, making a query to the database. It's used by Article.insertRecord() to create a new record for an article in the DB. This corresponds to the 'C' in CRUD, since it CREATES a new entry in the DB.
+  let SQL = `
+    INSERT INTO articles(title, author, author_url, category, published_on, body)
+    VALUES ($1, $2, $3, $4, $5, $6);
+  `;
+
+  let values = [
+    request.body.title,
+    request.body.author,
+    request.body.author_url,
+    request.body.category,
+    request.body.published_on,
+    request.body.body
+  ]
+
+  client.query( SQL, values )
+    .then(function() {
+      response.send('insert complete')
+    })
+    .catch(function(err) {
+      console.error(err);
+    });
+});
+
+app.put('/articles/:id', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This also matches #3 & #5 in the diagram, since it queries the DB and responds to the client. It's used by Article.updateRecord to update a record in the DB. This matches the 'U' in CRUD, since it's updating the record.
```

Comment:
This function isn't serving data up to the front end, So 5 is incorrect. Take a look at the diagram again and think about whats happening when you update a record.

Reply:
The function does actually send a response to the front end - 'update complete', but I assume the comment was supposed to be about the block that contains the query.